src/lucky/tags/text_helpers.cr

```diff
@@ -0,0 +1,244 @@
+        text.rindex(separator, length_with_room_for_omission) || length_with_room_for_omission
+      else
+        length_with_room_for_omission
+      end
+
+    "#{text[0, stop]}#{omission}"
+  end
+
+  def highlight(text : String, phrases : Array(String | Regex), highlighter : Proc | String = "<mark>\\1</mark>")
+    if text.blank? || phrases.all?(&.to_s.blank?)
+      text || ""
+    else
+      match = phrases.map do |p|
+        p.is_a?(Regex) ? p.to_s : Regex.escape(p.to_s)
+      end.join("|")
+
+      if highlighter.is_a?(Proc)
+        text.gsub(/(#{match})(?![^<]*?>)/i, &highlighter)
+      else
+        text.gsub(/(#{match})(?![^<]*?>)/i, highlighter)
+      end
+    end
+  end
+
+  def highlight(text : String, phrases : Array(String | Regex), &block : String -> _)
+    highlight(text, phrases, highlighter: block)
+  end
+
+  def highlight(text : String, phrase : String | Regex, highlighter : Proc | String = "<mark>\\1</mark>")
+    phrases = [phrase] of String | Regex
+    highlight(text, phrases, highlighter)
+  end
+
+  def highlight(text : String, phrase : String | Regex, &block : String -> _)
+    phrases = [phrase] of String | Regex
+    highlight(text, phrases, highlighter: block)
+  end
+
+  def excerpt(text : String, phrase : Regex | String, separator : String = "", radius : Int32 = 100, omission : String = "...")
+    return "" if text.to_s.blank?
+
+    case phrase
+    when Regex
+      regex = phrase
+    else
+      regex = /#{Regex.escape(phrase.to_s)}/i
+    end
+
+    return unless matches = text.match(regex)
+    phrase = matches[0]
+
+    unless separator.empty?
+      text.split(separator).each do |value|
+        if value.match(regex)
+          phrase = value
+          break
+        end
+      end
+    end
+
+    first_part, second_part = text.split(phrase, 2)
+
+    prefix, first_part   = cut_excerpt_part(:first, first_part, separator, radius, omission)
+    postfix, second_part = cut_excerpt_part(:second, second_part, separator, radius, omission)
+
+    affix = [first_part, separator, phrase, separator, second_part].join.strip
+    [prefix, affix, postfix].join
+  end
+
+  def pluralize(count : Int32 | String | Nil, singular : String, plural = nil)
+    word = if (count == 1 || count =~ /^1(\.0+)?$/)
+      singular
+    else
+      plural || LuckyInflector::Inflector.pluralize(singular)
+    end
+
+    "#{count || 0} #{word}"
+  end
+
+  def word_wrap(text : String, line_width : Int32 = 80, break_sequence : String = "\n")
+    text = text.split("\n").map do |line|
+      line.size > line_width ? line.gsub(/(.{1,#{line_width}})(\s+|$)/, "\\1#{break_sequence}").strip : line
+    end
+    text.join(break_sequence)
+  end
+
+  def simple_format(text : String, **html_options, wrapper_tag : String | Symbol = "p")
+    paragraphs = split_paragraphs(text)
+
+    paragraphs = [""] if paragraphs.empty?
+
+    paragraphs.map { |paragraph|
+      content_tag(wrapper_tag, paragraph, **html_options)
+    }.join("\n\n")
```

Comment:
The version without a block could do something like this to remove duplicate code:

```crystal
def simple_format(text : String)
  simple_format(text) do |formatted_text|
    para formatted_text
  end
end
```

What do you think?

Reply:
Fails a few tests

```crystal
view.simple_format("We want to put a wrapper...\n\n...right there."){ |txt| view.div txt }.should eq "<div>We want to put a wrapper...</div>\n\n<div>...right there.</div>"
```
```
       Expected: "<div>We want to put a wrapper...</div>\n\n<div>...right there.</div>"
            got: "\n\n"
```

```crystal
view.simple_format(""){ |txt| view.div txt }.should eq "<div></div>"
```
```
       Expected: "<div></div>"
            got: ""
```

```crystal
view.simple_format("crazy\r\n cross\r platform linebreaks").should eq "<p>crazy\n<br /> cross\n<br /> platform linebreaks</p>"
```
```
       Expected: "<p>crazy\n<br /> cross\n<br /> platform linebreaks</p>"
            got: "<p>crazy\n&lt;br /&gt; cross\n&lt;br /&gt; platform linebreaks</p>"
```

Reply:
For the first spec I think you need to call `to_s` on the result since it returns an `IO::Memory` now and not a String. The second on I think should be an empty string since there is no text, so there shouldn't be a wrapper for the paragraphs. I think for the third one it would be a bit trickier. I'm not totally sure how to handle that one. 

I think for now we should remove the option to use a custom block. Instead let's do just `para` as the default, remove the capture and content_tag stuff and see if we can figure out a way to do it with the block later. That way it doesn't hold up the rest of this PR for too long. What do you think?

Reply:
I prefer the working software ;-) You really that against content_tag and capture? Can we make them private to contain their use? I have a sneaking suspicion they are useful for other purposes.

If we don't want to hold up the other helper I can break them into their own files.

Reply:
Yeah working software is good haha. I am pretty against content_tag for now (but maybe add it in the future as a separate PR), but I'm pretty against capture at this point. It may be useful, but I'd rather wait to add it. I'm fairly certain this feature can be done without it, but I don't have time to dive into it right now :( 

I'd be ok with separating the simple_format stuff into their own PR. This one is a bit big so that would make it easier to review as well 👍